test/dst.test.ts

```diff
@@ -0,0 +1,173 @@
+      console.log(indent + "├── ");
+      printTree(tree.children[i], indent + "│   ");
+    }
+    console.log(indent + "└── ");
+    printTree(tree.children[tree.children.length - 1], indent + "    ");
+  }
+}
+
+describe("Simulate failures", () => {
+  test("Simulate failure 1", async () => {
+    const baseline = new Resonate();
+    baseline.register("test", test1);
+
+    const { context: context1, promise: promise1 } = baseline._run("test", "baseline");
+    await promise1;
+
+    // Map context to something useable
+    const tree1 = traces(context1);
+    printTree(tree1);
+
+    const storedContexts: Context[] = [context1];
+
+    const probFailure = 0.8;
+    console.log("simulated probability of failure: " + probFailure);
+
+    const continueLoop = true;
+    while (continueLoop) {
+      const resonate = new Resonate();
+
+      resonate.register("test", test1);
+      try {
+        const testRandomSeed = Math.random();
+        console.log("testRandomSeed: " + testRandomSeed);
+        const { context: currentContext, promise: currentPromise } = resonate._run("test", "baseline", {
+          test: probFailure,
+          testRandomSeed: testRandomSeed,
+        });
+
+        await currentPromise;
+        // Break the loop if the current context was successful
+        const tree2 = traces(currentContext);
+        printTree(tree2);
+
+        storedContexts.push(currentContext);
+        break;
+      } catch (e) {
+        console.log("Failed to run test, trying again!");
+      }
+    }
+    expect(isSubsetTree(context1, storedContexts[1])).toBe(true);
+  });
+
+  test("Simulate failure 2", async () => {
+    const baseline = new Resonate();
+    baseline.register("test", test2);
+
+    const { context: context1, promise: promise1 } = baseline._run("test", "baseline");
+    await promise1;
+
+    // Map context to something useable
+    const tree1 = traces(context1);
+    printTree(tree1);
+
+    const storedContexts: Context[] = [context1];
+
+    const probFailure = 0.8;
+    console.log("simulated probability of failure: " + probFailure);
+
+    const continueLoop = true;
+    while (continueLoop) {
+      const resonate = new Resonate();
+
+      resonate.register("test", test2);
+      try {
+        const testRandomSeed = Math.random();
+        console.log("testRandomSeed: " + testRandomSeed);
+        const { context: currentContext, promise: currentPromise } = resonate._run("test", "baseline", {
+          test: probFailure,
+          testRandomSeed: testRandomSeed,
+        });
+
+        await currentPromise;
+        // Break the loop if the current context was successful
+        const tree2 = traces(currentContext);
+        printTree(tree2);
+
+        storedContexts.push(currentContext);
+        break;
+      } catch (e) {
+        console.log("Failed to run test, trying again!");
+      }
+    }
+    expect(isSubsetTree(context1, storedContexts[1])).toBe(true);
```

Comment:
Is it possible to clean up the console.logs and only print information after this statement if the expect fails? If so let's clean up the console.logs in this file

Reply:
done